### PR TITLE
Prepare 0.4.0 with attestation.

### PR DESCRIPTION
**Pull request by jsirois**
Also update dependencies.

Work towards #64.

### CODE REVIEW

File: docs/manifest.md [hunk 1]
```diff
@@ -2,16 +2,13 @@
...
-allow_raw_typenames: "no"
-
-# TODO(John Sirois): https://github.com/a-scie/lift/issues/32
-#  Flip to "no" and work through errors until the manifest format is fully documented.
-allow_missing_doc: "yes"
-
-link_tables: "yes"
-recurse_tables: "yes"
----
+:toctree_maxdepth: 2
```
Comment (jsirois) [inline docs/manifest.md:9]: The Myst update warns about the deprecated directive option syntax and comment syntax (and our doc build treats warnings as errors); so this updates to the approved syntax for both options directives and comments.

File: noxfile.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def run_pex(session: Session, script, *args, silent=False, **env) -> Any | None:
         *args,
         env={"PEX_SCRIPT": script, **env},
         silent=silent,
-        stderr=subprocess.DEVNULL if silent else None,
+        stderr=subprocess.DEVNULL if silent else subprocess.STDOUT,
```
Comment (jsirois) [inline noxfile.py:46]: Better type-checking flagged this.

File: docs/_ext/sphinx_science/directives.py [hunk 3]
```diff
@@ -60,7 +61,7 @@ class DirectiveSpec:
     name: str
     args: tuple[str, ...] = ()
     options: Mapping[str, str] = FrozenDict()
-    content: str = ""
+    content: StringList = field(default_factory=StringList)
```
Comment (jsirois) [inline docs/_ext/sphinx_science/directives.py:64]: Updated docutils necessitated this change.

File: docs/_ext/sphinx_science/directives.py [hunk 4]
```diff
@@ -133,7 +134,7 @@ def register(
 
         class Synthesized(Directive):
             has_content = getattr(doc_gen_directive, "has_content", False)
-            option_spec: Mapping[str, Callable[[str], Any]] = {
+            option_spec = {
```
Comment (jsirois) [inline docs/_ext/sphinx_science/directives.py:137]: Better type-checking allowed this manual type ascription to be dropped.

**Review by benjyw — APPROVED** [review]

**Pull request merged**